src/main/java/com/amihaiemil/zold/RtWallet.java

```diff
@@ -143,12 +146,24 @@ public void pay(
     }
 
     @Override
-    public JsonArray find(
-            final String id, final String details
-    ) throws IOException, URISyntaxException {
+    public JsonArray find(final Map<String, String> params)
+            throws IOException, URISyntaxException {
         URIBuilder builder = new URIBuilder(this.baseUri.toString() + "/find");
-        builder.setParameter("bnf", id);
-        builder.setParameter("details", details);
+
+        Set<String> validParams = new HashSet<>();
```

Comment:
@ammaratef45 it can be converted to static field:
```java
private static final Set<String> VALID = Collections.unmodifiableSet(
            new HashSet<>(
                Arrays.asList(
                    "id", "date", "amount", "prefix", "bnf", "details"
                )
            )
        );